server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1477,7 +1478,8 @@ public NumberFieldType(
             Map<String, String> meta,
             FieldValues<Number> script,
             boolean isDimension,
-            MetricType metricType
+            MetricType metricType,
+            IndexMode indexMode
```

Comment:
I wish there was a better way to implement this without leaking the index mode into so many field mappers.

Reply:
I agree that this is kind of annoying. I don't see another way and this happened in other field mappers as well.